pkg/operator/target_status_test.go

```diff
@@ -1251,9 +1251,11 @@ func TestPolling(t *testing.T) {
 	expectStatus := func(t *testing.T, description string, expected []monitoringv1.ScrapeEndpointStatus) {
 		// Must poll because status is updated via other thread.
 		var err error
-		if pollErr := wait.Poll(100*time.Millisecond, 2*time.Second, func() (done bool, err error) {
+		if pollErr := wait.Poll(100*time.Millisecond, 2*time.Second, func() (bool, error) {
 			var podMonitorings monitoringv1.PodMonitoringList
-			kubeClient.List(ctx, &podMonitorings)
+			if err := kubeClient.List(ctx, &podMonitorings); err != nil {
+				return false, nil
```

Comment:
nit: By returning `nil` here, we're effectively nacking any errors when doing the list. Is that the desired behavior? Or should we error and exit immediately in the test?